misk-aws/src/main/kotlin/misk/jobqueue/sqs/SqsJobQueue.kt

```diff
@@ -15,10 +16,20 @@ import javax.inject.Singleton
 internal class SqsJobQueue @Inject internal constructor(
   private val queues: QueueResolver,
   private val metrics: SqsMetrics,
-  private val tracer: Tracer
+  private val tracer: Tracer,
+  private val tokenGenerator: TokenGenerator
 ) : JobQueue {
+
+  override fun enqueue(
+    queueName: QueueName,
+    body: String,
+    deliveryDelay: Duration?,
+    attributes: Map<String, String>
```

Comment:
worth validating that idempotency key isn't provided?

Reply:
good point. should probably also namespace the key like `_misk_idempotency_key`